File: Python/2-Medium/00038._外观数列/00038._外观数列.py
def count_and_say(n: int) -> str:
    """
    生成外观数列的第 n 项

    参数:
    n -- 外观数列的项数

    返回:
    外观数列的第 n 项字符串

    示例:
    >>> count_and_say(1)
    '1'
    >>> count_and_say(4)
    '1211'
    """
    # 初始化第一项
    current = "1"
    
    # 对于每一项进行迭代
    for _ in range(1, n):
        next_seq = ""  # 准备下一项
        i = 0  # 段落开始位置
        
        # 遍历当前项
        while i < len(current):
            count = 1  # 计数器
            # 统计相同数字的连续长度
            while i + 1 < len(current) and current[i] == current[i + 1]:
                i += 1
                count += 1
            
            # 构建下一项
            next_seq += str(count) + current[i]
            i += 1
        
        # 准备下一轮迭代
        current = next_seq
    
    # 返回最终结果
    return current

# 测试代码
print(count_and_say(1))  # 应输出: '1'
print(count_and_say(4))  # 应输出: '1211'
print(count_and_say(5))  # 测试更大的n，验证逻辑正确性

File: C++/2-Medium/00038._外观数列/00038._外观数列.cpp
#include <iostream>
#include <string>
using namespace std;

// 生成外观数列的下一项
string getNext(const string &current) {
    string result;
    int count = 1; // 计数器，至少有一个字符
    // 从第一个字符开始遍历
    for (int i = 1; i <= current.length(); ++i) {
        // 如果当前字符与前一个相同，计数器增加
        if (i < current.length() && current[i] == current[i - 1]) {
            count++;
        } else {
            // 否则，添加前一个字符的计数和字符本身到结果字符串
            result += to_string(count) + current[i - 1];
            count = 1; // 重置计数器
        }
    }
    return result;
}

// 生成外观数列第 n 项
string countAndSay(int n) {
    string current = "1";
    // 从第二项开始，生成外观数列，直到第 n 项
    for (int i = 2; i <= n; ++i) {
        current = getNext(current);
    }
    return current;
}

int main() {
    // 测试代码
    int n = 4; // 可以修改 n 的值以测试不同的输入
    cout << "The " << n << "th term of the count-and-say sequence is: " << countAndSay(n) << endl;
    return 0;
}